site/src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  ReactElement,
+  ReactNode,
+  cloneElement,
+  createContext,
+  useContext,
+  useRef,
+  useState,
+} from "react";
+// This is used as base for the main Popover component
+// eslint-disable-next-line no-restricted-imports -- Read above
+import MuiPopover, {
+  type PopoverProps as MuiPopoverProps,
+} from "@mui/material/Popover";
+
+type TriggerMode = "hover" | "click";
+
+type TriggerRef = React.RefObject<HTMLElement>;
+
+type TriggerElement = ReactElement<{
+  onClick?: () => void;
+  ref: TriggerRef;
+}>;
+
+type PopoverContextValue = {
+  open: boolean;
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  triggerRef: TriggerRef;
+  mode: TriggerMode;
+};
+
+const PopoverContext = createContext<PopoverContextValue | undefined>(
+  undefined,
+);
+
+export const Popover = (props: {
+  children: ReactNode | ((popover: PopoverContextValue) => ReactNode); // Allows inline usage
+  mode?: TriggerMode;
+  defaultOpen?: boolean;
+}) => {
+  const [open, setOpen] = useState(props.defaultOpen ?? false);
+  const triggerRef = useRef<HTMLElement>(null);
+  const value = { open, setOpen, triggerRef, mode: props.mode ?? "click" };
+
+  return (
+    <PopoverContext.Provider value={value}>
+      {typeof props.children === "function"
+        ? props.children(value)
+        : props.children}
+    </PopoverContext.Provider>
+  );
+};
+
+export const usePopover = () => {
+  const context = useContext(PopoverContext);
+  if (!context) {
+    throw new Error(
+      "Popover compound components cannot be rendered outside the Popover component",
+    );
+  }
+  return context;
+};
+
+export const PopoverTrigger = (props: {
+  children: TriggerElement;
+  hover?: boolean;
+}) => {
+  const popover = usePopover();
+
+  const clickProps = {
+    onClick: () => {
+      popover.setOpen((open) => !open);
+    },
+  };
+
+  const hoverProps = {
+    onMouseEnter: () => {
+      popover.setOpen(true);
+    },
+    onMouseLeave: () => {
+      popover.setOpen(false);
+    },
+  };
+
+  return cloneElement(props.children, {
```

Comment:
Actually, I thought of one more suggestion for the trigger: is there a way for you to merge in the `aria-haspopup` prop in as well, just to make sure that the trigger is getting picked up properly by screen readers?

Reply:
Sure, I will take a look and see how to add them.

Reply:
Do you have good material about that? I'm not sure how to make this happen 🤔

Reply:
I actually saw it in MUI's documentation – it's in the source code for the hover popover example:
https://mui.com/material-ui/react-popover/

I'm not sure how the React.clone operation would need to be changed, but two properties would need to be added in:
```tsx
{
  "aria-owns": isOpen ? "some-id-value" : undefined,
  "aria-haspopup": true
}
```
But I can also take a look into this later today